lib/greet.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+module.exports = exports = {};
+var name = process.argv[2];
+
+exports.greet = function(name) {
+  if (arguments.length === 0) throw new Error('name not entered');
+  console.log(`hello ${name}!`);
+  return `hello ${name}!`;
+};
+
+exports.greet(name);
```

Comment:
Since you are exporting a single function, module.exports = exports = { } isn't necessary in this case. You would want to use module.exports = { } to export a single function. For instance, if you attached another method onto the exports object then they way you are using module.exports on line 3 would be just fine :)

Also, since you are using module.exports there's no need to explicitly export the function again on line 12. You could just omit that entirely.